src/mockoptimizer.jl

```diff
@@ -216,21 +152,113 @@ mutable struct MockLinQuadOptimizer <: LQOI.LinQuadOptimizer
     params::Dict{String,Any}
     l_rows::Vector{Int}
     q_rows::Vector{Int}
+
+    termination_status_stored::Vector{MOI.TerminationStatusCode}
+    primal_status_stored::Vector{MOI.ResultStatusCode}
+    dual_status_stored::Vector{MOI.ResultStatusCode}
+
+    variable_primal_solution_stored::Vector{Vector{Float64}}
+    variable_dual_solution_stored::Vector{Vector{Float64}}
+
+    constraint_primal_solution_stored::Vector{Vector{Float64}}
+    constraint_dual_solution_stored::Vector{Vector{Float64}}
+
+    quadratic_primal_solution_stored::Vector{Vector{Float64}}
+    quadratic_dual_solution_stored::Vector{Vector{Float64}}
+
     MockLinQuadOptimizer(::Nothing) = new()
 end
 
 LQOI.LinearQuadraticModel(::Type{MockLinQuadOptimizer},env::Nothing) = MockLinQuadModel(env,"defaultname")
 
 function MockLinQuadOptimizer(;kwargs...)
 
-    m = MockLinQuadOptimizer(nothing)
-    m.params = Dict{String,Any}()
-    MOI.empty!(m)
+    instance = MockLinQuadOptimizer(nothing)
+    instance.params = Dict{String,Any}()
+    instance.l_rows = Int[]
+    instance.q_rows = Int[]
+
+    instance.termination_status_stored = MOI.TerminationStatusCode[]
+    instance.primal_status_stored = MOI.ResultStatusCode[]
+    instance.dual_status_stored = MOI.ResultStatusCode[]
+
+    instance.variable_primal_solution_stored = Vector{Float64}[]
+    instance.variable_dual_solution_stored = Vector{Float64}[]
+
+    instance.constraint_primal_solution_stored = Vector{Float64}[]
+    instance.constraint_dual_solution_stored = Vector{Float64}[]
+
+    instance.quadratic_primal_solution_stored = Vector{Float64}[]
+    instance.quadratic_dual_solution_stored = Vector{Float64}[]
+
+    MOI.empty!(instance)
     for (name,value) in kwargs
-        m.params[string(name)] = value
-        setparam!(m.inner, string(name), value)
+        instance.params[string(name)] = value
+        setparam!(instance.inner, string(name), value)
     end
-    return m
+    return instance
+end
+
+function unload(from,to,warn = true)
+    if !isempty(from)
+        out = from[1]
+        try
+            out = copy(from[1])
+        end
+        shift!(from)
+        return out
+    else
+        if warn
+            Compat.@warn("cant solve this model no extra solution")
```

Comment:
Warnings and errors should be properly formatted English sentences. I don't know what that warning means.